### PR TITLE
Remove reflection and boxed math operations #37

### LINKED ISSUE #37: Reflection warnings
I've compiled my program that uses guardrails with 'warn-on-reflection' and 'warn-on-boxed-math' set to true. It appears that guardrails utilises a significant amount of reflection, resulting in these warnings appearing in my program. This is crucial, as I compile everything with GraalVM native-image, and minimising reflection is preferable.

### PR DESCRIPTION
**Pull request by kwojcik-blockether**
Fixes #37 

Found some more warnings.

### CODE REVIEW

File: src/main/com/fulcrologic/guardrails/config.cljc [hunk 1]
```diff
@@ -105,6 +105,11 @@
           (catch Exception _
             (require '[com.fulcrologic.guardrails.stubs.cljs-env :as cljs-env]))))
 
+#?(:clj
```
Comment (awkay) [inline src/main/com/fulcrologic/guardrails/config.cljc]: I do not want to propagate this to every user.

File: src/main/com/fulcrologic/guardrails/core.cljc [hunk 2]
```diff
@@ -10,19 +10,23 @@
...
+              [com.fulcrologic.guardrails.impl.pro :as gr.pro]
+              [com.fulcrologic.guardrails.utils :as utils :refer [cljs-env? clj->cljs strip-colors]]]
+       :cljs [[com.fulcrologic.guardrails.impl.externs]
+              [com.fulcrologic.guardrails.utils :as utils :refer [strip-colors]]
+              [goog.object :as gobj]])
+   [clojure.core.async :as async]
+   [clojure.string :as str]
+   [expound.alpha :as exp]))
+
+#?(:clj
```
Comment (awkay) [inline src/main/com/fulcrologic/guardrails/core.cljc]: Again, this cannot be left here.

File: src/main/com/fulcrologic/guardrails/impl/externs.cljc [hunk 3]
```diff
@@ -4,6 +4,11 @@
     [com.fulcrologic.guardrails.registry :as gr.reg]
     [com.fulcrologic.guardrails.utils :as utils]))
 
+#?(:clj
```
Comment (awkay) [inline src/main/com/fulcrologic/guardrails/impl/externs.cljc]: another one

File: src/main/com/fulcrologic/guardrails/impl/pro.cljc [hunk 4]
```diff
@@ -5,6 +5,11 @@
     [com.fulcrologic.guardrails.registry :as gr.reg]
     [com.fulcrologic.guardrails.utils :as utils]))
 
+#?(:clj
```
Comment (awkay) [inline src/main/com/fulcrologic/guardrails/impl/pro.cljc]: ditto

File: src/main/com/fulcrologic/guardrails/malli/core.cljc [hunk 5]
```diff
@@ -14,6 +14,10 @@
     [malli.error :as me]
     [malli.registry]))
 
+#?(:clj
```
Comment (awkay) [inline src/main/com/fulcrologic/guardrails/malli/core.cljc]: ditto

File: src/main/com/fulcrologic/guardrails/malli/registry.cljc [hunk 6]
```diff
@@ -4,6 +4,10 @@
     [malli.core :as m]
     [malli.registry :as mr]))
 
+#?(:clj
+   (do
+     (set! *warn-on-reflection* true)
+     (set! *unchecked-math* :warn-on-boxed)))
```
Comment (awkay) [inline src/main/com/fulcrologic/guardrails/malli/registry.cljc]: ditto

File: src/main/com/fulcrologic/guardrails/utils.cljc [hunk 7]
```diff
@@ -9,12 +9,29 @@
...
-    [clojure.string :as str]
-    [clojure.walk :as walk]))
+   #?(:clj [clojure.stacktrace :as st])
+   [clojure.string :as str]
+   [clojure.walk :as walk])
+  #?(:clj
+     (:import
+      [java.lang Throwable])))
+
+#?(:clj
```
Comment (awkay) [inline src/main/com/fulcrologic/guardrails/utils.cljc]: ditto

File: src/main/com/fulcrologic/guardrails/utils.cljc [hunk 8]
```diff
@@ -262,28 +279,30 @@
...
+      (aset backtrace 0 (mod (inc next-entry) sz)))))
 
 (defn backtrace-exit []
   (when *backtrace*
-    (let [next-entry  (aget *backtrace* 0)
-          sz          (dec (count *backtrace*))
-          prior-entry (mod (dec next-entry) sz)]
-      (aset *backtrace* (+ 1 prior-entry) empty-entry)
-      (aset *backtrace* 0 prior-entry))))
+    (let [^"[Ljava.lang.Object;" backtrace *backtrace*
```
Comment (awkay) [inline src/main/com/fulcrologic/guardrails/utils.cljc]: inconsistent use of hint-backtrace.

**Review by awkay — CHANGES_REQUESTED** [review]

**Comment by kwojcik-blockether** [comment]
@awkay thank you for all the comments. It's ready to be merged! now.

**Review by awkay — APPROVED** [review]

**Comment by awkay** [comment]
Thanks

**Pull request merged**